tests/unit/plugins/callback/test_opentelemetry.py

```diff
@@ -0,0 +1,49 @@
+# (C) 2021, Victor Martinez <VictorMartinezRubio@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
```

Comment:
Work in progress, though any hints how I can run this script locally rather than waiting for the CI to validate it?

Thanks

Reply:
Thanks Felix, though I'm still struggling with this, I managed to clone ansible and prepare the terminal context but 

1) I faced some issues with the folder structure:

```
ERROR: The current working directory must be at or below:

 - the Ansible source: /Users/vmartinez/work/src/github.com/v1v/ansible/
 - an Ansible collection: {...}/ansible_collections/{namespace}/{collection}/
```

So I renamed my project from `/Users/victor/work/src/github.com/v1v/community.general` to `/Users/victor/work/src/github.com/v1v/ansible_collections/v1v/community.general`

and I don't see that error anymore

2) I faced issues with `ansible-test units --docker -v tests/unit/plugins/callback/test_opentelemetry.py`

```
E   ModuleNotFoundError: No module named 'ansible_collections.v1v.community'
- generated xml file: /root/ansible_collections/v1v/community.general/tests/output/junit/python3.8-controller-units.xml -
```

I even renamed my folder from `/Users/victor/work/src/github.com/v1v/ansible_collections/v1v/community.general` to `/Users/victor/work/src/github.com/v1v/ansible_collections/ansible-collections/community.general`

But I see the same error :/ 


```
==================================== ERRORS ====================================
____________________ ERROR collecting test_opentelemetry.py ____________________
ImportError while importing test module '/root/ansible_collections/ansible-collections/community.general/tests/unit/plugins/callback/test_opentelemetry.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ModuleNotFoundError: No module named 'ansible_collections.ansible-collections.community'
- generated xml file: /root/ansible_collections/ansible-collections/community.general/tests/output/junit/python3.8-controller-units.xml -
=========================== short test summary info ============================
ERROR tests/unit/plugins/callback/test_opentelemetry.py
=============================== 1 error in 0.60s ===============================
ERROR: Command "pytest --boxed -r a -n auto --color yes -p no:cacheprovider -c /root/ansible/test/lib/ansible_test/_data/pytest.ini --junit-xml /root/ansible_collections/ansible-collections/community.general/tests/output/junit/python3.8-controller-units.xml --strict-markers -v tests/unit/plugins/callback/test_opentelemetry.py" returned exit status 1.
```

Any ideas whether I missed something that was needed?

Thanks

Reply:
Finally, I've managed to find the required folder structure:

- `ansible_collections/community/general`, those details were in https://github.com/ansible-collections/community.general/blob/3502f3b48690500c8d60942ca85bb3527f856e4e/tests/utils/shippable/shippable.sh#L68

I can now proceed with the local testing :)